pennylane/qnn/keras.py

```diff
@@ -426,6 +427,42 @@ def _evaluate_qnode(self, x):
 
         return res
 
+    def construct(self, args, kwargs):
+        """Constructs the wrapped QNode on input data using the initialized weights.
+
+        This method was added to match the QNode interface. The provided args
+        must contain a single item, which is the input to the layer. The provided
+        kwargs is unused.
+
+        Args:
+            args (tuple): A tuple containing one entry that is the input to this layer
+            kwargs (dict): Unused
+        """
+        # GradientTape required to ensure that the weights show up as trainable on the qtape
+        with tf.GradientTape() as tape:
+            tape.watch(list(self.qnode_weights.values()))
+
+            inputs = args[0]
+            kwargs = {self.input_arg: inputs, **{k: 1.0 * w for k, w in self.qnode_weights.items()}}
+            self.qnode.construct((), kwargs)
+
+    def __getattr__(self, item):
+        """If the given attribute does not exist in the class, look for it in the wrapped QNode."""
```

Comment:
it's more the other way around - if it's not in the wrapped QNode (or if it's not initialized), look in the class. also, would `getattr(self, item)` be wrong below? Same applies for similar code elsewhere in this PR. But I guess the answer depends on my question re: `_initialized`

Reply:
The way Python works is that `__getattr__` is only called if `obj.some_attr` raises an `AttributeError`, so there's no way to look in the `QNode` first. Also for methods like `__call__`, we definitely want to look in this class first before the `QNode`.

Reply:
right! a rather quirky dunder, makes sense.